c10/core/ScalarType.h

```diff
@@ -77,7 +79,8 @@ namespace c10 {
   _(double, Double, d)            \
   _(c10::qint8, QInt8, i)         \
   _(c10::quint8, QUInt8, i)       \
-  _(c10::qint32, QInt32, i)
+  _(c10::qint32, QInt32, i)       \
+  _(at::BFloat16, BFloat16, d)
```

Comment:
wait, why doesn't this have bool?

Reply:
AT_FORALL_SCALAR_TYPES(_) is not being used anywhere in our code base and i was actually going to remove BFloat16 from here as we have this macro for BC reasons. 
Should i add bool or should i remove BFloat16?

Reply:
might be a question for @izdeby ?

Reply:
if it's not being used anywhere, can't you just kill it?  (You can do this in a different PR if you want).